### ISSUE TITLE
v6.4.0 throws when used with postcss-discard-comments v2.0.4

### ISSUE THREAD

**Issue by jesstelford** [postcss/autoprefixer#708]
I have created a minimal test-case here: https://gist.github.com/jesstelford/3cdebdee4691ce5fdd4ae696db7f296f

This css:

```css
.spaceship {
  transform: translateX(0) ;
}
```

Will throw the below error:

```
  TypeError: Cannot read property 'constructor' of null
    at clone (/home/teddy/dev/postcss-autoprefixer-bug/node_modules/autoprefixer/lib/prefixer.js:13:21)
    at clone (/home/teddy/dev/postcss-autoprefixer-bug/node_modules/autoprefixer/lib/prefixer.js:31:19)
    at clone (/home/teddy/dev/postcss-autoprefixer-bug/node_modules/autoprefixer/lib/prefixer.js:31:19)
    at Function.Prefixer.clone (/home/teddy/dev/postcss-autoprefixer-bug/node_modules/autoprefixer/lib/prefixer.js:64:16)
    at TransformDecl.Prefixer.clone (/home/teddy/dev/postcss-autoprefixer-bug/node_modules/autoprefixer/lib/prefixer.js:112:23)
    at TransformDecl.Declaration.insert (/home/teddy/dev/postcss-autoprefixer-bug/node_modules/autoprefixer/lib/declaration.js:105:30)
    at TransformDecl.insert (/home/teddy/dev/postcss-autoprefixer-bug/node_modules/autoprefixer/lib/hacks/transform-decl.js:64:47)
    at TransformDecl.Declaration.add (/home/teddy/dev/postcss-autoprefixer-bug/node_modules/autoprefixer/lib/declaration.js:138:19)
    at TransformDecl.Prefixer.process (/home/teddy/dev/postcss-autoprefixer-bug/node_modules/autoprefixer/lib/prefixer.js:104:18)
    at TransformDecl.Declaration.process (/home/teddy/dev/postcss-autoprefixer-bug/node_modules/autoprefixer/lib/declaration.js:144:50)
```

When executed like so:

```
const processor = postcss();
processor.use(discardComments());
processor.use(autoprefixer());
processor.process(css)
```

If postcss-discard-comments is turned off, prefixing works again. It seems like autoprefixer isn't able to handle some transformation that discard-comments performs.

**Comment by ai** [comment]
Seems like AST could be broken :(

Could you create a test project, please?

**Comment by jesstelford** [comment]
@ai Already did, it's linked at the very top: https://gist.github.com/jesstelford/3cdebdee4691ce5fdd4ae696db7f296f

**Comment by ai** [comment]
Thanks. I hope this issue is not so critical, because I could look at it only in weekend.

You can use two PostCSS engines to fix it right now:

```js
.pipe( gulpPostcss([autoprefixer]) )
.pipe( gulpPostcss([discardComments]) )
```

**Comment by jesstelford** [comment]
Excellent workaround, thanks :)

Take your time - I'm not even sure it's an autoprefixer bug (could be discard-comments causing the misstep).

**Comment by ai** [comment]
Fixed 01e34a7
